packages/ra-core/src/controller/create/CreateContextProvider.tsx

```diff
@@ -38,7 +38,12 @@ export const CreateContextProvider = ({
     value: CreateControllerResult;
 }) => (
     <CreateContext.Provider value={value}>
-        <SaveContextProvider value={usePickSaveContext(value)}>
+        <SaveContextProvider
+            value={{
+                ...usePickSaveContext(value),
+                mutationMode: 'pessimistic',
```

Comment:
Why don't you get the mutationMode inside `usePickSaveContext`?

Reply:
Because I can't tell if I'm being called from a CreateContext ('pessimistic') or an EditContext ('value.mutationMode')

Reply:
You can still do it in `usePickSaveContext` but override it for create like you already did